h/api/auth.py

```diff
@@ -90,3 +92,26 @@ def userid_from_bearer_token(request):
                           algorithms=['HS256']).get('sub')
     except InvalidTokenError:
         return None
+
+
+def userid_from_dev_token(request):
+    """
+    Return the userid authenticated by the given request's dev token.
+
+    If the request contains a developer API token in its Authorization header
+    then return the userid corresponding to the token.
+
+    If the request has no Authorization header or has an Authorization header
+    that isn't a valid dev token then return None.
+
+    :param request: the request to return the authenticated userid for
+    :type request: pyramid.request.Request
+
+    :returns: the userid authenticated by the request's dev token, or None
+    :rtype: unicode or None
+    """
+    if not request.headers.get('Authorization', '').startswith('dev-'):
+        return None
+
+    token = unicode(request.headers['Authorization'])
+    return models.Token.userid_for_token(token)
```

Comment:
Not sure about adding this helper method to the Token model. This seems like it would be just as clear:

```python
tok = models.Token.query.get(value)
if tok is None:
    return None
return tok.userid
```